core/server/proxy/src/main/java/alluxio/web/ProxyWebServer.java

```diff
@@ -108,6 +112,18 @@ public ProxyWebServer(String serviceName, InetSocketAddress address,
         .register(S3RestExceptionMapper.class);
 
     mFileSystem = FileSystem.Factory.create(Configuration.global());
+    // Lazy creating metadata directory. Do not create directory directly,
+    // this will change the behavior of constructor and cause some tests to fail.
+    mMultipartUploadsMetadataDirCreator = new Runnable() {
+      private final AtomicBoolean mIsCreated = new AtomicBoolean(false);
```

Comment:
Constructing a Runnable task here feels very strange, why not just init multipart dir and set a global flag directly here?

Reply:
I will do that, thanks.